Created simple readme based on my current understanding.

neonWhiteout opened the pull request.

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
 [![Tests](https://github.com/AlgebraicJulia/StockFlow.jl/actions/workflows/tests.yml/badge.svg)](https://github.com/AlgebraicJulia/StockFlow.jl/actions/workflows/tests.yml)
 <!-- TODO: Set up on codecov.io for repo [![codecov](https://codecov.io/github/AlgebraicJulia/StockFlow.jl/branch/maaster/graph/badge.svg)](https://app.codecov.io/github/AlgebraicJulia/StockFlow.jl) -->
+
+StockFlow.jl is a Julia library for creating stock and flow diagrams using category theory.  Stockflows are used to represent systems where its population can move between different states, such as tracking how many people have been infected or recovered from a disease.  By providing initial values, and values for the parameters, we can solve the differential equation which the stockflow represen ...
```
For 'category theory', might it be helpful to link to the research and mention early on in the README some of the benefits of this model, like composition and stratification? On the WIP I wrote up, I had

> StockFlow.jl is a Julia library for the creation and execution of `[stock and flow modeling diagrams](https://en.wikipedia.org/wiki/System_dynamics#Stock_and_flow_diagrams)`, designed with model composition and stratification in mind through a `[categorical approach](https://arxiv.org/abs/2211.01290)`.

I think we want people that visit the library to know right away what it's about and why it's neat.

On the note of linking https://arxiv.org/abs/2211.01290 -- @Xiaoyan-Li thoughts on linking publications in the README for additional information?

Xiaoyan-Li replied:
@Saityi yes, I agree with your suggestions. https://arxiv.org/abs/2205.08373 and https://arxiv.org/abs/2211.01290  are two papers we have worked on related to the current implement. I would definitely suggest to include to those two papers.

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
 [![Tests](https://github.com/AlgebraicJulia/StockFlow.jl/actions/workflows/tests.yml/badge.svg)](https://github.com/AlgebraicJulia/StockFlow.jl/actions/workflows/tests.yml)
 <!-- TODO: Set up on codecov.io for repo [![codecov](https://codecov.io/github/AlgebraicJulia/StockFlow.jl/branch/maaster/graph/badge.svg)](https://app.codecov.io/github/AlgebraicJulia/StockFlow.jl) -->
+
+StockFlow.jl is a Julia library for creating stock and flow diagrams using category theory.  Stockflows are used to represent systems where its population can move between different states, such as tracking how many people have been infected or recovered from a disease.  By providing initial values, and values for the parameters, we can solve the differential equation which the stockflow represen ...
```
Would it also be a good idea to phrase this using more common terminology e.g. instead of 'Stockflows' maybe 'Stock and flow diagrams' or, from the paper, 'stock-flow diagrams'?

neonWhiteout replied:
Yeah, I just used 'Stockflow' for brevity.

neonWhiteout replied:
Maybe we should mention causal loop diagrams as well?

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
+
+Stockflows can be composed and stratified - that is, combined on stocks and sum variables or split into subpopulations.  One can also do algebraic rewriting for more manual, specific fixes, such as preventing a dynamic variable from unnecessarily being computed twice, or substituting one parameter for another.
+
+Examples of the domain specific language, composition, stratification and algebraic rewriting can be seen in the examples folder.  In particular, full\_fledged\_schema\_examples\_new contains examples which use the domain specific language.
```
I think you can link [examples](examples) from the README using `[examples](examples)` and similar for more specific examples.

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
+Stockflows can be composed and stratified - that is, combined on stocks and sum variables or split into subpopulations.  One can also do algebraic rewriting for more manual, specific fixes, such as preventing a dynamic variable from unnecessarily being computed twice, or substituting one parameter for another.
+
+Examples of the domain specific language, composition, stratification and algebraic rewriting can be seen in the examples folder.  In particular, full\_fledged\_schema\_examples\_new contains examples which use the domain specific language.
+
```
In the WIP README I had, I also had a code example and a specific note that this library does not have an official release and is under active development, below between `---`s:

---

## Example interpretation of a stock and flow diagram using an ODE solver

From the [SEIR Composition Example](examples/full_fledged_schema_examples/composition/SEIR_full_model_measles_chickenpox.ipynb):

``` julia
seir = @stock_and_flow begin
    :stocks
    S
    E
    I
    R

    :parameters
    μ
    β
    tlatent
    trecovery
    δ

    :flows
    ☁ => fbirth(μ * N) => S
    S => fincid(β * S * I / N) => E
    S => fdeathS(S * δ) => ☁
    E => finf(E / tlatent) => I
    E => fdeathE(E * δ) => ☁
    I => frec(I / trecovery) => R
    I => fdeathI(I * δ) => ☁
    R => fdeathR(R * δ) => ☁

    :sums
    N = [S, E, I, R]

end

# define parameter values and initial values of stocks
# define constant parameters
p_measles = LVector(
    β=49.598, μ=0.03/12, δ=0.03/12, tlatent=8.0/30, trecovery=5.0/30
)

# define initial values for stocks
u0_measles = LVector(
    S=90000.0-930.0, E=0.0, I=930.0, R=773545.0
)

prob_measles = ODEProblem(vectorfield(seir),u0_measles,(0.0,120.0),p_measles);
sol_measles = solve(prob_measles,Tsit5(),abstol=1e-8);
plot(sol_measles)
```

See the full example for additional comments and the chickenpox model.

## Note

This library is under active development and does not yet have an official release.

---

thoughts on something like this in the README?

neonWhiteout replied:
I like it.  Might want to include an example of explicitly defining a dynamic variable?
I'm not entirely sure what length we're aiming for here.

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
+
+StockFlow.jl is a Julia library for creating stock and flow diagrams using category theory.  Stockflows are used to represent systems where its population can move between different states, such as tracking how many people have been infected or recovered from a disease.  By providing initial values, and values for the parameters, we can solve the differential equation which the stockflow represen ...
+
+In this particular schema, stockflows consist of stocks, flows, dynamic variables, sum variables, parameters, and the links between them.
```
'stockflows consist' => 'a stock and flow diagram consists ...' or 'stock-flow diagrams consist'?

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
+In this particular schema, stockflows consist of stocks, flows, dynamic variables, sum variables, parameters, and the links between them.
+
+* Stocks represent accumulations of a population in a particular state, such as 'susceptible', 'infected', 'recovered', etc.
+* Flows represent transitions between states.  Flow can go from a stock to a stock, from nothing to a stock, or from a stock to nothing, the latter two representing a population entering or leaving a model.  Flows are dependent on a dynamic variable, which determines its rate.
```
'A flow can go from a stock to a stock' sounds better to me.

Saityi commented on README.md:
```diff
@@ -1,3 +1,20 @@
+* Sum variables represent the sum of a subpopulation.  The value of a sum variable is the sum of all the stocks which link to it.  Common examples include the sum of the entire population N, the sum of infected variables NI, or the sum of a particular subpopulation NS.
+* Parameters are variables for which the particular values are defined outside the model definition.  You can provide parameters and initial values, then solve the differential equation.
+
+Stockflows can be created and manipulated using StockFlow and Syntax, the latter of which includes the domain specific language for easier creation.
```
'using `StockFlow` and `StockFlow.Syntax` modules' might be a more specific phrasing than 'using StockFlow and Syntax'.

Also, another 'stockflows'.

neonWhiteout commented:
Incorporated your suggestions
* Replaced stockflows with stock-flow diagrams
* Included your first paragraph
* Included your example at the bottom (also included a comment about defining dynamic variables implicitly or with :dynamic_variables
* Minor other syntax touchups, like replacing Syntax with StockFlow.Syntax and surrounding with code blocks, linking examples, etc.

Saityi reviewed (approved):
LGTM. Nice work!

neonWhiteout commented:
Just saw small typo, in the example flows it should be => rather than =
